include/kernel_structs.h

```diff
@@ -181,8 +181,7 @@ extern struct z_kernel _kernel;
  */
 bool z_smp_cpu_mobile(void);
 
-#define _current_cpu ({ __ASSERT_NO_MSG(!z_smp_cpu_mobile()); \
-			arch_curr_cpu(); })
```

Comment:
This is a correctness check, and it's absolutely needed otherwise it's very easy to use _current_cpu incorrectly.  For clarity: the reason we have this is because receiving an interrupt can suspend the current thread, and it may be resumed on a different CPU.  So otherwise the evaluation of arch_curr_cpu() can return a pointer value that is **incorrect** by the time it is used.